Microsoft.Toolkit.Wpf.UI.XamlHost/UwpUIElementExtensions.cs

```diff
@@ -2,24 +2,47 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-namespace Microsoft.Toolkit.Wpf.UI.XamlHost
-{
-    /// <summary>
-    /// Extensions for use with UWP UIElement objects wrapped by the WindowsXamlHostBaseExt
-    /// </summary>
-    public static class UwpUIElementExtensions
-    {
-        private static Windows.UI.Xaml.DependencyProperty WrapperProperty { get; } =
-            Windows.UI.Xaml.DependencyProperty.RegisterAttached("Wrapper", typeof(System.Windows.UIElement), typeof(UwpUIElementExtensions), new Windows.UI.Xaml.PropertyMetadata(null));
-
-        public static WindowsXamlHostBase GetWrapper(this Windows.UI.Xaml.UIElement element)
-        {
-            return (WindowsXamlHostBase)element.GetValue(WrapperProperty);
-        }
-
-        public static void SetWrapper(this Windows.UI.Xaml.UIElement element, WindowsXamlHostBase wrapper)
-        {
-            element.SetValue(WrapperProperty, wrapper);
-        }
-    }
-}
+using System;
+
+namespace Microsoft.Toolkit.Wpf.UI.XamlHost
+{
+    /// <summary>
+    /// Extensions for use with UWP UIElement objects wrapped by the WindowsXamlHostBaseExt
+    /// </summary>
+    public static class UwpUIElementExtensions
+    {
+        private static bool IsDesktopWindowsXamlSourcePresent() => Windows.Foundation.Metadata.ApiInformation.IsApiContractPresent("Windows.UI.Xaml.Hosting.HostingContract", 3);
+
+        private static Windows.UI.Xaml.DependencyProperty WrapperProperty
+        {
+            get
+            {
+                if (IsDesktopWindowsXamlSourcePresent())
+                {
+                    var result = Windows.UI.Xaml.DependencyProperty.RegisterAttached("Wrapper", typeof(System.Windows.UIElement), typeof(UwpUIElementExtensions), new Windows.UI.Xaml.PropertyMetadata(null));
+                    return result;
+                }
+
+                throw new NotImplementedException();

```

Comment:
>NotImplementedException [](start = 26, length = 23)

Where does this exception get caught (so the designer still functions)?

Reply:
Yes.

---
In reply to: [259030180](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/pull/81#discussion_r259030180) [](ancestors = 259030180)